generator/transit_generator.cpp

```diff
@@ -125,15 +146,39 @@ void DeserializeGatesFromJson(my::Json const & root, string const & mwmDir, stri
   }
 }
 
+template <class Item>
+bool IsValid(vector<Item> const & items)
+{
+  return all_of(items.cbegin(), items.cend(), [](Item const & i) { return i.IsValid(); });
+}
+
 /// \brief Reads from |root| (json) and serializes an array to |serializer|.
 template <class Item>
 void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer)
 {
   vector<Item> items;
   DeserializeFromJson(root, key, items);
-
+  CHECK(IsValid(items), ("key:", key, "items:", items));
   serializer(items);
 }
+
+/// \brief Updates |edges| by adding valid value for Edge::m_weight if it's not valid.
+/// \note Edge::m_stop1Id and Edge::m_stop2Id should be valid for every edge in |edges| before call.
+void CalculateEdgeWeight(vector<Stop> const & stops, vector<transit::Edge> & edges)
+{
+  CHECK(is_sorted(stops.cbegin(), stops.cend(), LessById), ());
```

Comment:
Why do you use CHECK here and ASSERTion above?

Reply:
In short the reason is efficiency.

On the one hand I prefer to use CHECKs. Especially in generator. In most cases release build of generate_tool is launch so ASSERT calls are not seen. On the other hand I don't want the soft goes slow down because of it.

The method (`CalculateEdgeWeight `) is called once per mwm build. So I guess it's ok to call is_sorted in release (CHECK).
Method `Stop const & FindStopById(vector<Stop> const & stops, StopId stopId)` is called twice for every edge and the number of edges may be relevantly big in mwm. I'd prefer not to call is_sorted to often.

Reply:
Up to you.